Data/src/test/java/org/tribuo/data/columnar/RowProcessorTest.java

```diff
@@ -170,4 +183,127 @@ public void metadataExtractorTest() {
         assertThrows(PropertyException.class, () -> new RowProcessor<>(badExtractors,weightExtractor,response,fixed,Collections.emptySet()));
     }
 
+    @Test
+    public void replaceNewlinesWithSpacesTest() {
+        final Pattern BLANK_LINES = Pattern.compile("(\n[\\s-]*\n)+");
+
+        final Function<CharSequence, CharSequence> newLiner = (CharSequence charSequence) -> {
+            if (charSequence == null || charSequence.length() == 0) {
+                return charSequence;
+            }
+            return BLANK_LINES.splitAsStream(charSequence).collect(Collectors.joining(" ⏎\n\n"));
+        };
+
+        Tokenizer tokenizer = new MungingTokenizer(new BreakIteratorTokenizer(Locale.US), newLiner);
+        TokenPipeline textPipeline = new TokenPipeline(tokenizer, 2, false);
+
+        final Map<String, FieldProcessor> fieldProcessors = new HashMap<>();
+        fieldProcessors.put("order_text", new TextFieldProcessor("order_text", textPipeline));
+
+        MockResponseProcessor response = new MockResponseProcessor("Label");
+
+        Map<String,String> row = new HashMap<>();
+        row.put("order_text", "Jimmy\n\n\n\nHoffa");
+        row.put("Label", "Sheep");
+
+        RowProcessor<MockOutput> processor = new RowProcessor<>(Collections.emptyList(),null,response,fieldProcessors,Collections.emptyMap(),Collections.emptySet(), false);
+
+        Example<MockOutput> example = processor.generateExample(row,true).get();
+
+        // Check example is extracted correctly
+        assertEquals(5,example.size());
+        assertEquals("Sheep",example.getOutput().label);
+        Iterator<Feature> featureIterator = example.iterator();
+        Feature a = featureIterator.next();
+        assertEquals("order_text@1-N=Hoffa", a.getName());
+        assertEquals(1.0, a.getValue());
+        a = featureIterator.next();
+        assertEquals("order_text@1-N=Jimmy", a.getName());
+        a = featureIterator.next();
+        assertEquals("order_text@1-N=⏎", a.getName());
+        a = featureIterator.next();
+        assertEquals("order_text@2-N=Jimmy/⏎", a.getName());
+        a = featureIterator.next();
+        assertEquals("order_text@2-N=⏎/Hoffa", a.getName());
+        assertFalse(featureIterator.hasNext());
+        a = featureIterator.next(); // weird this doesn't throw NoSuchElementException?
+        assertNull(a.getName());
+    }
+
+    public abstract static class ForwardingTokenizer implements Tokenizer {
```

Comment:
Could this be folded into `MungingTokenizer`? It doesn't seem to be used anywhere else.

Reply:
sure, i just copy+pasted in from my project

Reply:
i wonder if those might be useful to others?

Reply:
Well we can consider expanding the `Tokenizer` implementations in a different PR, but for the purposes of the test lets minimise the code that's added.

Reply:
👍  done